plugins/calendar/cdcalendarcontroller.cpp

```diff
@@ -44,8 +45,13 @@ void updateNotebooks(AccountId id, bool enabled)
         AccountId accountInt = accountStr.toULong(&ok);
         if (ok && accountInt == id) {
             bool visible = notebook->isVisible();
-            if (visible != enabled) {
-                notebook->setIsVisible(enabled);
+            if (!enabled && visible) {
+                notebook->setIsVisible(false);
+                notebook->setCustomProperty(VISIBILITY_CHANGED_FLAG, QString::fromLatin1("true"));
+                storage->updateNotebook(notebook);
+            } else if (enabled && !visible & !notebook->customProperty(VISIBILITY_CHANGED_FLAG).isEmpty()) {
```

Comment:
```suggestion
            } else if (enabled && !visible && !notebook->customProperty(VISIBILITY_CHANGED_FLAG).isEmpty()) {
```
I'm just wondering if this might be clearer and just a little safer?

Reply:
Of course, my bad. Thanks for spotting it.